test/test.js

```diff
@@ -2080,6 +2081,45 @@ describe("follow-redirects", function () {
         });
     });
   });
+
+  describe("when request is going through an HTTP proxy (without a tunnel)", function() {
+    [
+      { redirectType: "absolute", redirectUrl: "http://localhost:3600/b" },
+      { redirectType: "relative", redirectUrl: "/b" },
+    ].forEach(function (testCase) {
+      it("redirects to proper URL when Location header is " + testCase.redirectType, function() {
+        app.get("/a", redirectsTo(testCase.redirectUrl));
+        app.get("/b", sendsJson({ good: "yes" }));
+        app2.port = 3601;
+        app2.all("*", proxy("localhost:3601"));
```

Comment:
Note: using `express-http-proxy` wasn't practical here because it's meant to forward requests to a determined host - it doesn't implement an HTTP proxy as per the RFC.